enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,461 @@
+
+Since indexes are manipulated by making additive changes to it, the way an index can be reproduced today is my managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) and by managing a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, and thereby has significantly burdened the `opm` tool with feature requests to add an increasing number of sub-commands to allow index authors to manipulate their indexes in various ways. 
+
+### Goals
+
+A human consumable representation of packages in an index in the form of json/yaml will: 
+
+1. Allow for editing of package level information without rebuilding and republishing artifacts(remove bundle/s from a channel, switch bundle/s to a different channel etc).
+2. Allow algebraic operations to be performed on indexes in a seamless way(combine two indexes, split an index, remove bundles from the index, etc).
+3. Allow for validation of the structure of an index against a standard definition of an index.
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducibility of an index using just the yaml/json representation of the index. 
+
+### Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations.    
+
+### User Stories 
+
+#### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the packages of an index.
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --index=docker.io/my-namespace/community-operators --output=json`
+
```

Comment:
I would expect a "read" operation to write to the filesystem. Could this print a stream to stdout instead?

Reply:
@njhale do you think it gives out a "read" vibe because of `inspect`? Does it help if we use `render` instead? 
As for the stream, I don't see how that could be useful to the user, if they want to eventually make changes to the objects.